pkg/apis/flow/v1alpha1/jobflow_types.go

```diff
@@ -31,86 +31,129 @@ type JobFlowSpec struct {
 	// Important: Run "make" to regenerate code after modifying this file
 
 	// Foo is an example field of JobFlow. Edit jobflow_types.go to remove/update
-	Flows           []Flow `json:"flows,omitempty"`
-	JobRetainPolicy string `json:"jobRetainPolicy,omitempty"`
+	// +optional
+	Flows []Flow `json:"flows,omitempty"`
+	// +optional
+	JobRetainPolicy RetainPolicy `json:"jobRetainPolicy,omitempty"`
 }
 
 // Flow defines the dependent of jobs
 type Flow struct {
-	Name      string     `json:"name"`
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name"`
+	// +optional
 	DependsOn *DependsOn `json:"dependsOn,omitempty"`
 }
 
 type DependsOn struct {
+	// +optional
 	Targets []string `json:"targets,omitempty"`
-	Probe   *Probe   `json:"probe,omitempty"`
+	// +optional
+	Probe *Probe `json:"probe,omitempty"`
 }
 
 type Probe struct {
-	HttpGetList    []HttpGet    `json:"httpGetList,omitempty"`
-	TcpSocketList  []TcpSocket  `json:"tcpSocketList,omitempty"`
+	// +optional
+	HttpGetList []HttpGet `json:"httpGetList,omitempty"`
+	// +optional
+	TcpSocketList []TcpSocket `json:"tcpSocketList,omitempty"`
+	// +optional
 	TaskStatusList []TaskStatus `json:"taskStatusList,omitempty"`
 }
 
 type HttpGet struct {
-	TaskName   string        `json:"taskName,omitempty"`
-	Path       string        `json:"path,omitempty"`
-	Port       int           `json:"port,omitempty"`
+	// +optional
```

Comment:
Why have so many lines unrelated to this project been added?

Reply:
These commits is synced from master branch, will submit a seperate pr later.